src/store/inflight_activation.rs

```diff
@@ -627,39 +821,37 @@ impl InflightActivationStore {
     /// complete.
     #[instrument(skip_all)]
     pub async fn handle_failed_tasks(&self) -> Result<FailedTasksForwarder, Error> {
-        let mut atomic = self.write_pool.begin().await?;
+        let mut atomic = self.meta_write_pool.begin().await?;
 
-        let failed_tasks: Vec<SqliteRow> =
-            sqlx::query("SELECT activation FROM inflight_taskactivations WHERE status = $1")
-                .bind(InflightActivationStatus::Failure)
-                .fetch_all(&mut *atomic)
-                .await?
-                .into_iter()
-                .collect();
+        let failed_tasks: Vec<SqliteRow> = sqlx::query(
+            "SELECT id, on_attempts_exceeded FROM inflight_taskactivations WHERE status = $1",
+        )
+        .bind(InflightActivationStatus::Failure)
+        .fetch_all(&mut *atomic)
+        .await?
+        .into_iter()
+        .collect();
 
         let mut forwarder = FailedTasksForwarder {
             to_discard: vec![],
             to_deadletter: vec![],
         };
 
         for record in failed_tasks.iter() {
-            let activation_data: &[u8] = record.get("activation");
-            let activation = TaskActivation::decode(activation_data)?;
-
-            // Without a retry state, tasks are discarded
-            if activation.retry_state.as_ref().is_none() {
-                forwarder.to_discard.push(activation.id);
-                continue;
-            }
             // We could be deadlettering because of activation.expires
             // when a task expires we still deadletter if configured.
-            let retry_state = &activation.retry_state.as_ref().unwrap();
-            if retry_state.on_attempts_exceeded == OnAttemptsExceeded::Discard as i32
-                || retry_state.on_attempts_exceeded == OnAttemptsExceeded::Unspecified as i32
+            // let retry_state = &activation.retry_state.as_ref().unwrap();
+            let id: &str = record.get("id");
+            let infl_on_attempts_exceeded: InflightOnAttemptsExceeded =
+                record.get("on_attempts_exceeded");
```

Comment:
Nice :clap: even if the separate blob db doesn't work out we should make this change so that we scan/load fewer bytes.